### ISSUE TITLE
`Naming/InclusiveLanguage`'s `AllowedRegex` does not support free-spacing/extended mode Regexp

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#11831]
[Ruby's Regexp "free-spacing"/"extended" mode (`/.../x`)](https://docs.ruby-lang.org/en/master/Regexp.html#class-Regexp-label-Free-Spacing+Mode+and+Comments) allows the use of additional whitespace and comments, which is useful to express a complex or otherwise long regular expression.

These are (de)serializable as YAML, the format used in RuboCop configuration files.

--------

## Expected behavior

As such, it should be possible to use such a regular expression as configuration for the `Naming/InclusiveLanguage` cop's `AllowedRegex` config (and all other cops, for that matter).

Given the files listed in the reproduction section below:

```sh
$ bundle exec rubocop --only Naming/InclusiveLanguage
Inspecting 2 files
..

2 files inspected, no offenses detected
```

## Actual behavior

```
bundle exec rubocop --only Naming/InclusiveLanguage 
Inspecting 2 files
.C

Offenses:

example.rb:1:2: C: Naming/InclusiveLanguage: Consider replacing 'master' with another term.
:mastercard
 ^^^^^^
example.rb:2:2: C: Naming/InclusiveLanguage: Consider replacing 'master' with another term.
:mastermind
 ^^^^^^

2 files inspected, 2 offenses detected
```

<details><summary><code>--debug</code></summary>

```
bundle exec rubocop --debug --only Naming/InclusiveLanguage
For /Users/sambostock/src/local/rubocop_inclusive_language_free_spacing.rb: configuration from /Users/sambostock/src/local/rubocop_inclusive_language_free_spacing.rb/.rubocop.yml
Default configuration from /Users/sambostock/.gem/ruby/3.2.2/gems/rubocop-1.50.2/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/sambostock/.cache/rubocop_cache/7a2a65ad3891cc55333c8e84fe1fc49849c788ea/98f069e84e9f37415c1544e9767039eaea49508b/3d0d8b164e427f6266892b307ab7c8935e882152
Loading cache from /Users/sambostock/.cache/rubocop_cache/7a2a65ad3891cc55333c8e84fe1fc49849c788ea/98f069e84e9f37415c1544e9767039eaea49508b/f04a6c2e0a12bc07fad2192371c0cdf651355532
Inspecting 2 files
Scanning /Users/sambostock/src/local/rubocop_inclusive_language_free_spacing.rb/Gemfile
Loading cache from /Users/sambostock/.cache/rubocop_cache/7a2a65ad3891cc55333c8e84fe1fc49849c788ea/98f069e84e9f37415c1544e9767039eaea49508b/3d0d8b164e427f6266892b307ab7c8935e882152
.Scanning /Users/sambostock/src/local/rubocop_inclusive_language_free_spacing.rb/example.rb
Loading cache from /Users/sambostock/.cache/rubocop_cache/7a2a65ad3891cc55333c8e84fe1fc49849c788ea/98f069e84e9f37415c1544e9767039eaea49508b/f04a6c2e0a12bc07fad2192371c0cdf651355532
C

Offenses:

example.rb:1:2: C: Naming/InclusiveLanguage: Consider replacing 'master' with another term.
:mastercard
 ^^^^^^
example.rb:2:2: C: Naming/InclusiveLanguage: Consider replacing 'master' with another term.
:mastermind
 ^^^^^^

2 files inspected, 2 offenses detected
Finished in 0.33647400001063943 seconds
```

</details>

## Steps to reproduce the problem

```yaml
# .rubocop.yml
AllCops:
  DisabledByDefault: true

Naming/InclusiveLanguage:
  Enabled: true
  FlaggedTerms:
    master:
      AllowedRegex:
        - !ruby/regexp '/master(?:
            card|
            mind
          )/x'
        # - !ruby/regexp /master(?:card|mind)/
```

```ruby
# example.rb
:mastercard
:mastermind
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.50.2 (using Parser 3.2.2.1, rubocop-ast 1.28.0, running on ruby 3.2.2) [arm64-darwin22]
```

## Investigation

I haven't looked in detail, but I suspect this is due to the use of `.source` on the Regexp and manually combining the contents (which presumably doesn't preserve the `/.../x` flag).

https://github.com/rubocop/rubocop/blob/03ce5917fab15fda274ad984e23bfbb87fafb9b5/lib/rubocop/cop/naming/inclusive_language.rb#L189-L195

This might be solvable by using `Regexp.union` without resorting to `Regexp.source`.

**Comment by sambostock** [comment]
I have opened a draft PR with an approach which I think addresses this: https://github.com/rubocop/rubocop/pull/11832

If the approach is suitable, I can tidy it up.
